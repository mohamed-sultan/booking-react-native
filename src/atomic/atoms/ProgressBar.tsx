import React from 'react';
import {View, StyleSheet} from 'react-native';
import  { useThemeColors } from '@store/useTheme';

interface ProgressBarBaseProps {
  progress?: number;
}

const ProgressBarBase = ({
  progress = 0.25,
}: ProgressBarBaseProps) => {
  const colors = useThemeColors();

  const styles = getStyles(colors);
  return (
    <View style={styles.barBackground}>
      <View
        style={[
          styles.progressFill,
          {width: `${progress * 100}%`, backgroundColor: colors.PrimaryBackground},
        ]}
      />
    </View>
  );
};

const getStyles = (colors: any) => StyleSheet.create({
  barBackground: {
    height: 4,
    backgroundColor: colors.LightBackground,
    borderRadius: 4,
    overflow: 'hidden',
  },
  progressFill: {
    height: '100%',
  },
});

const ProgressBar = React.memo(ProgressBarBase);
export default ProgressBar;
