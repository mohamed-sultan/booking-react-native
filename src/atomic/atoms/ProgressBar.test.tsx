import React from 'react';
import renderer from 'react-test-renderer';
import ProgressBar from './ProgressBar';

// Mock useThemeColors
jest.mock('@store/useTheme', () => ({
  useThemeColors: () => ({
    LightBackground: '#e0e0e0',
    PrimaryBackground: '#00aaff',
  }),
}));

describe('ProgressBar', () => {
  it('renders correctly with default progress', () => {
    const tree = renderer.create(<ProgressBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with 0% progress', () => {
    const tree = renderer.create(<ProgressBar progress={0} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with 50% progress', () => {
    const tree = renderer.create(<ProgressBar progress={0.5} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with 100% progress', () => {
    const tree = renderer.create(<ProgressBar progress={1} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
