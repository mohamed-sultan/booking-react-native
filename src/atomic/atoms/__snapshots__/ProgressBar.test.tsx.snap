// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders correctly with 0% progress 1`] = `null`;

exports[`ProgressBar renders correctly with 50% progress 1`] = `null`;

exports[`ProgressBar renders correctly with 100% progress 1`] = `null`;

exports[`ProgressBar renders correctly with default progress 1`] = `null`;
